### PR TITLE
add eWeLink temp/humi Sensor

### PR DESCRIPTION
**Pull request by mingwei0827**

### CODE REVIEW

**Review by PKacprowiczS — APPROVED** [review]

File: devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy [hunk 1]
```diff
@@ -34,6 +34,9 @@ metadata {
 		fingerprint profileId: "0104",  inClusters: "0000, 0001, 0003, 0009, 0402", manufacturer: "HEIMAN", model: "HT-EM", deviceJoinName: "HEIMAN Multipurpose Sensor" //HEIMAN Temperature & Humidity Sensor
 		fingerprint profileId: "0104",  inClusters: "0000, 0001, 0003, 0402, 0B05", manufacturer: "HEIMAN", model: "HT-EF-3.0", deviceJoinName: "HEIMAN Multipurpose Sensor" //HEIMAN Temperature & Humidity Sensor
 		fingerprint profileId: "0104", deviceId: "0302", inClusters: "0000,0001,0003,0020,0402,0405", outClusters: "0003,000A,0019", manufacturer: "frient A/S", model :"HMSZB-110", deviceJoinName: "frient Multipurpose Sensor" // frient Humidity Sensor
+
+		//eWeLink
+		fingerprint profileId: "0104", inClusters: "0000, 0001, 0003, 0402, 0405",  outClusters: "0003", manufacturer: "eWeLink", model: "TH01", deviceJoinName: "eWeLink Temp/Humidity Sensor"
```
Comment (PKacprowiczS) [inline devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy]: change `deviceJoinName` to `"eWeLink Multipurpose Sensor"`

**Review by PKacprowiczS — CHANGES_REQUESTED** [review]

File: devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy (hunk 1 above)
Comment (mingwei0827) [reply devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy]: I have modified the file.

**Review by PKacprowiczS — DISMISSED** [review]
@greens

File: devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy [hunk 2]
```diff
@@ -214,3 +230,11 @@ def configure() {
...
 	device.getDataValue("manufacturer") == "frient A/S"
 }
+
+private boolean isEWeLink() {
+	device.getDataValue("manufacturer") == "eWeLink"
+}
+
+private boolean isEWeLinkTh01() {
+	isEWeLink() && device.getDataValue("model") == "TH01"
+}
```
Comment (greens) [inline devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy]: could we get rid of the separate `isWeLink` function?

Comment (mingwei0827) [reply devicetypes/smartthings/smartsense-temp-humidity-sensor.src/smartsense-temp-humidity-sensor.groovy]: hi~ I have modified the file.

**Review by greens — APPROVED** [review]

**Comment by greens** [comment]
will be live on the 27th

**Comment by teba-home** [comment]
Appologies if this is the incorrect place to post this. If so, please direct me where to post.
I am having issues with the snzb-02 showing offline in ST. It joined as eWeLink Multipurpose Sensor and is using the stock SmartSense Temp/Humidity Sensor DTH

**Comment by k-morawski** [comment]
> Appologies if this is the incorrect place to post this. If so, please direct me where to post.
> I am having issues with the snzb-02 showing offline in ST. It joined as eWeLink Multipurpose Sensor and is using the stock SmartSense Temp/Humidity Sensor DTH

Dear @mingwei0827 , could you please check this? Many thanks, Kamil

**Comment by mingwei0827** [comment]
hi Kamil

Which issues are you talking about?



------------------------------------------------------------------
发件人：k-morawski ***@***.***>
发送时间：2021年9月1日(星期三) 17:32
收件人：SmartThingsCommunity/SmartThingsPublic ***@***.***>
抄　送：杨明炜 ***@***.***>; Mention ***@***.***>
主　题：Re: [SmartThingsCommunity/SmartThingsPublic] add eWeLink temp/humi Sensor (#70669)

Appologies if this is the incorrect place to post this. If so, please direct me where to post.
 I am having issues with the snzb-02 showing offline in ST. It joined as eWeLink Multipurpose Sensor and is using the stock SmartSense Temp/Humidity Sensor DTH
Dear @mingwei0827 , could you please check this? Many thanks, Kamil
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by k-morawski** [comment]
> hi Kamil Which issues are you talking about?
> […](#)
> ------------------------------------------------------------------ 发件人：k-morawski ***@***.***> 发送时间：2021年9月1日(星期三) 17:32 收件人：SmartThingsCommunity/SmartThingsPublic ***@***.***> 抄　送：杨明炜 ***@***.***>; Mention ***@***.***> 主　题：Re: [SmartThingsCommunity/SmartThingsPublic] add eWeLink temp/humi Sensor (#70669) Appologies if this is the incorrect place to post this. If so, please direct me where to post. I am having issues with the snzb-02 showing offline in ST. It joined as eWeLink Multipurpose Sensor and is using the stock SmartSense Temp/Humidity Sensor DTH Dear @mingwei0827 , could you please check this? Many thanks, Kamil — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

Hi @mingwei0827 , please refer to @teba-home comment: https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/70669#issuecomment-909283069
Because we do not have devices we cant test it on our side.
Many thanks,
Kamil

**Comment by mingwei0827** [comment]
> Appologies if this is the incorrect place to post this. If so, please direct me where to post.
> I am having issues with the snzb-02 showing offline in ST. It joined as eWeLink Multipurpose Sensor and is using the stock SmartSense Temp/Humidity Sensor DTH

hi teba
I suggest you send an email to support@itead.cc 
Ask questions and describe the problem in detail, so as to troubleshoot the problem efficiently.

**Pull request merged**